src/main/java/reactor/pool/Pool.java

```diff
@@ -112,4 +113,39 @@
                 PooledRef::release);
     }
 
+    /**
+     * Shutdown the pool by:
+     * <ul>
+     *     <li>
+     *         notifying every acquire still pending that the pool has been shut down,
+     *         via a {@link RuntimeException}
```

Comment:
Probably a `IllegalStateException` (or a dedicated `PoolShutdownException`) would improve debugging experience. A raw `RuntimeException` doesn't feel appropriate.

Reply:
yes, I'll introduce pool-specific exceptions in another PR I think.